practiceCourseGenerator/practiceCourseGenerator.py

```diff
@@ -0,0 +1,233 @@
+fh.setLevel(logging.INFO)
+
+#create console handler
+ch = logging.StreamHandler()
+ch.setLevel(logging.INFO)
+
+#create formatter and add it to the handlers
+formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+fh.setFormatter(formatter)
+ch.setFormatter(formatter)
+
+#add the handlers to the logger
+logger.addHandler(fh)
+logger.addHandler(ch)
+
+logger.info('Script Initiated')
+
+#def getInputFile():
+lines = []
+userSets = []
+users = []
+finalUsers = []
+userStrings = []
+dataSets = []
+directoryCounter = 0
+errorCount = 0
+outputDirectory = './canvas_files'
+userHeader = 'user_id,login_id,password,first_name,last_name,email,status'
+courseHeader = 'course_id,short_name,long_name,account_id,term_id,status,start_date,end_date'
+enrollmentHeader = 'course_id,user_id,role,section_id,status,associated_user_id'
+userDictionaries = []
+
+#Read properties file
+with open('propertiesProd.json') as data_file:    
+    data = json.load(data_file)
+
+logger.debug('urlPrefix: ' + str(data['URL_PREFIX']))
+logger.debug('driveFile: ' + str(data['DRIVE_FILE']))
+
+urlPrefix = str(data['URL_PREFIX'])
+urlPost = '/profile?access_token='
+
+#Download exceptions file
+CLIENT_SECRET = 'tl_client_secret.json'
+SCOPES = [
+	'https://www.googleapis.com/auth/drive.readonly',
+	'https://www.googleapis.com/auth/drive',
+	'https://www.googleapis.com/auth/drive.appdata',
+	'https://www.googleapis.com/auth/drive.apps.readonly',
+	'https://www.googleapis.com/auth/drive.file',
+	'https://www.googleapis.com/auth/drive.readonly'
+]
+store = file.Storage('storage.json')
+creds = store.get()
+if not creds or creds.invalid:
+    flow = client.flow_from_clientsecrets(CLIENT_SECRET, ' '.join(SCOPES))
+    creds = tools.run(flow, store)
+DRIVE = build('drive', 'v2', http=creds.authorize(Http()))
+file_Id = str(data['DRIVE_FILE'])
+gFile = DRIVE.files().get(fileId = file_Id).execute()
+download_file(DRIVE, gFile)
+
+#Get Token
+with open('token.txt') as f:
+    for line in f:
+    	token = line.rstrip('\n')
+
+#Get Input File
+with open('input.csv') as f:
+	for line in f:
+		line = line.rstrip('\n')
+		lines.append(line)
+
+for line in lines:
+	#skips header row
+	if line == lines[0]:
+			continue
+	logger.info('Line: ' + line)
+	parsedLine = line.split('"')
+	userString = parsedLine[1]
+	logger.info('User String: ' + userString)
+	userSets.append(userString)
+
+logger.info('Folder Exists: ' + str(os.path.isdir(outputDirectory)))
+
+if os.path.isdir(outputDirectory):
+	#rename
+	logger.info('Need to rename directory')
+	os.rename(outputDirectory,outputDirectory+'_'+logdate)
+	os.mkdir(outputDirectory)
+else:
+	#create 
+	logger.info('Need to create the directory')
+	os.mkdir(outputDirectory)
+
+urlPost = urlPost + token
+
+for userSet in userSets:
+	logger.info('User Set:')
+	users = (userSet.split(','))
+	for user in users:
+		if user not in finalUsers:
+			finalUsers.append(user)
+
+#Generate users.csv file
```

Comment:
I think these three loops to generate the CSVs are a bit repetitive.  I suggest generalizing them and moving them into functions.  Since each one has special requirements for each file, put that code in small formatting/processing functions that the CSV creator function can call as callbacks.  Or possibly do all the processing first, then pass them to the function that will put that data in the CSV.